inference-engine/tests/functional/inference_engine/ir_serialization/const_compression.cpp

```diff
@@ -0,0 +1,263 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <fstream>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include "ie_core.hpp"
+#include "gtest/gtest.h"
+
+#include <ngraph/function.hpp>
+#include <transformations/serialize.hpp>
+
+#ifndef IR_SERIALIZATION_MODELS_PATH // should be already defined by cmake
+#define IR_SERIALIZATION_MODELS_PATH ""
+#endif
+
+class SerializatioConstantCompressionTest : public ::testing::Test {
+protected:
+    std::string test_name =
+        ::testing::UnitTest::GetInstance()->current_test_info()->name();
+    std::string m_out_xml_path_1 = test_name + "1" + ".xml";
+    std::string m_out_bin_path_1 = test_name + "1" + ".bin";
+
+    void TearDown() override {
+        std::remove(m_out_xml_path_1.c_str());
+        std::remove(m_out_bin_path_1.c_str());
+    }
+
+    std::uintmax_t file_size(std::ifstream &f) {
+        // get length of file:
+        f.seekg(0, f.end);
+        std::uintmax_t length = f.tellg();
+        f.seekg(0, f.beg);
+        return length;
+    }
+};
+
+TEST_F(SerializatioConstantCompressionTest, IdenticalConstantsI32) {
+    std::shared_ptr<ngraph::Function> ngraph_a;
+    const int unique_const_count = 1;
+    ngraph::Shape shape{2, 2, 2};
+
+    auto A = ngraph::op::Constant::create(ngraph::element::i32, shape,
+        {1, 2, 3, 4, 5, 6, 7, 8});
+    auto B = ngraph::op::Constant::create(ngraph::element::i32, shape,
+        {1, 2, 3, 4, 5, 6, 7, 8});
+
+    ngraph_a = std::make_shared<ngraph::Function>(ngraph::NodeVector{A, B},
+        ngraph::ParameterVector{});
+
+    ngraph::pass::Serialize(m_out_xml_path_1, m_out_bin_path_1).run_on_function(ngraph_a);
+
+    std::ifstream xml_1(m_out_xml_path_1, std::ios::in | std::ios::binary);
+    std::ifstream bin_1(m_out_bin_path_1, std::ios::in | std::ios::binary);
```

Comment:
`std::ifstream` is always `std::ios::in`.
https://en.cppreference.com/w/cpp/io/basic_ifstream/basic_ifstream